src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -49,6 +49,10 @@ public function __construct(ObjectManager $manager, string $class, IdReader $idR
     {
         $classMetadata = $manager->getClassMetadata($class);
 
+        if ($idReader && !$idReader->isSingleId()) {
+            @trigger_error(sprintf('Passing an instance of "%s" with an entity class "%s" that have a composite id is deprecated since 4.3 and will throw an exception in 5.0.', IdReader::class, $class), E_USER_DEPRECATED);
```

Comment:
I don't get this, how do you know before hand that the entity has composite id without check `$idReader->isSingleId()`?

Reply:
See https://github.com/symfony/symfony/pull/30962 description.